pkg/operation/botanist/component/etcd/etcd_test.go

```diff
@@ -1491,6 +1491,94 @@ var _ = Describe("Etcd", func() {
 				})
 			})
 		})
+
+		Context("When etcd cluster is hibernated", func() {
+			BeforeEach(func() {
+				secretNamesToTimes := map[string]time.Time{}
+
+				var err error
+				sm, err = secretsmanager.New(
+					ctx,
+					logr.New(logf.NullLogSink{}),
+					testclock.NewFakeClock(time.Now()),
+					fakeClient,
+					testNamespace,
+					"",
+					secretsmanager.Config{
+						SecretNamesToTimes: secretNamesToTimes,
+					})
+				Expect(err).ToNot(HaveOccurred())
+
+				// Create new etcd CA
+				_, err = sm.Generate(ctx,
+					&secretsutils.CertificateSecretConfig{Name: v1beta1constants.SecretNameCAETCD, CommonName: "etcd", CertType: secretsutils.CACert})
+				Expect(err).ToNot(HaveOccurred())
+
+				// Create new peer CA
+				_, err = sm.Generate(ctx,
+					&secretsutils.CertificateSecretConfig{Name: v1beta1constants.SecretNameCAETCDPeer, CommonName: "etcd-peer", CertType: secretsutils.CACert})
+				Expect(err).ToNot(HaveOccurred())
+			})
+			JustBeforeEach(func() {
+				etcd = New(log, c, testNamespace, sm, Values{
+					Role:                    testRole,
+					Class:                   class,
+					Replicas:                pointer.Int32(0),
+					StorageCapacity:         storageCapacity,
+					StorageClassName:        &storageClassName,
+					DefragmentationSchedule: &defragmentationSchedule,
+					CARotationPhase:         gardencorev1beta1.RotationCompleted,
+					PriorityClassName:       priorityClassName,
+				})
+			})
+			It("Should not remove peer URL secrets", func() {
```

Comment:
Can you also add a test that if the Peer URL is nil, and etcd is hibernated then it should remain nil?

Reply:
Added a new test case
PTAL